### PR TITLE
Require config keys have default

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

Context:
The `habitat/config/default.py` config is important to the user since it contains all the defaults of the configurations loaded with `habitat.get_config`. As such, it is important that this file contains ALL possible keys the various tasks of habitat may use. This PR is meant to ensure that ALL keys that will be added to config files are present in the default (enforced in CI). Some config file can be manually excluded if they contain non-default keys on purpose.

Changes:
 - A new unit test has been added to make sure that any `.yaml` file in the `config` folder contains only keys that are present in the default config `habitat/config/default.py`.
 - In addition to the test, some keys of configs have been renamed for consistency across files.
 - Keys present in the configs that were not in the default have been added

Important:
 - The current default values contain my best guesses, if a value needs to be changed, please leave a comment.
 - The default config file is organized based off my best effort. If there is a better way to organize the keys, please leave a comment.
 - The default config of the *baselines* was left unchanged in this PR and will be modified in a subsequent one.
 - Some of these changes anticipate changes for habitat 2.0. Some of the keys in `default` are not *yet* on main.
 - There should be a way to make sure there are no unnecessary keys in the config file (not in this PR).

## How Has This Been Tested

The change has been unit tested. 

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by aclegg3 — COMMENTED** [review]
Thanks for cleaning up, @vincentpierre. Good to have a standard expectation for defaults, definitions, and param documentation.

File: habitat/config/default.py
Comment (vincentpierre) [reply habitat/config/default.py:75]: I do not know what a lot of these configurations do, so I don't think I can comment on them. A lot of these keys come from #774 and are not yet on main, I will wait for #774 to be merged before merging this one.

File: habitat/config/default.py
Comment (mathfac) [inline habitat/config/default.py:519]: @aclegg3, @ASzot is that right default value for Habitat Lab? It looks relevant only for Habitat 2.0 tasks and possibly need to be disabled by default. As well as other values like step physics, update robot, concur render?

Comment (mathfac) [reply habitat/config/default.py:519]: For some reason we get error for example_test on CI:
```
[17:26:07:066416]:[Physics] PhysicsManager.cpp(244)::addObject : Simplified template handle : 003_cracker_box  | newObjectHandle : 003_cracker_box_:0000
GL::Context: unsupported version string: 3.0.0 NVIDIA 418.87.00
BackgroundRenderer: Failed to create OpenGL context
```

File: test/test_config.py
Comment (Skylion007) [inline test/test_config.py:64]: glob already returns a list so this will be a list of list. This seems like a bug?
```suggestion
] + glob("**/pddl/*.yaml", recursive=True)

File: habitat/config/default.py
Comment (vincentpierre) [reply habitat/config/default.py]: The culprit was CONCUR_RENDER that I set to True instead of False. 
I changed it and the test is now passing. 
I also changed AUTO_SLEEP to False by default since the code seems to be [using False by default](https://github.com/facebookresearch/habitat-lab/blob/main/habitat/tasks/rearrange/rearrange_sim.py#L50).

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
This PR really makes me wish we had completed that migration to a better config system like Hydra. These options just keep growing.

**Review by mathfac — APPROVED** [review]

**Review by aclegg3 — APPROVED** [review]
LGTM. Thanks for this, much cleaner to validate configs this way.

After we merge this @ASzot, please take a close look at the defaults before merging back into `hab_suite_baselines` to be sure all these parameters reflect the expected values. Seems like a possible risk of conflicting with recent fixes there.

**Pull request merged**
